### DISCUSSION TITLE
SoM overlay implementation

### DISCUSSION THREAD

**Discussion by davidkaczer** [ServiceNow/BrowserGym#127] [Q&A] [answered]
The function `overlay_som` in `core.src.browsergym.utils.obs` implements SoM annotation for VisualWebArena, but it is currently unused by the agent loop. So it is currently not possible to fully replicate VisualWebArena.

Are there plans to implement this in the near future?

**Comment by gasse** [comment, marked as answer]
Hi @davidkaczer . This function is not called by default on the environment side, but your agent can import it and use it. See #115 #116

**Reply by davidkaczer** [reply]
Thanks for the reference!
